### PR TITLE
Add alignment-summary process

### PR DESCRIPTION
**Pull request by AnzeLovse**
## Checklist

* [x] Update CHANGELOG.rst for each commit separately:
  * Pay attention to write entries under the "Unreleased" section.
  * Mark all breaking changes as "**BACKWARD INCOMPATIBLE:**" and put them
    before non-breaking changes.
  * If a commit modifies a feature listed under "Unreleased" section,
    it might be sufficient to modify the existing CHANGELOG entry from previous
    commit(s).
* [x] Bump the process version:
  * **MAJOR version (first number)**: Backward incompatible changes (changes
    that break the api/interface). Examples: renaming the input/output, adding
    mandatory input, removing input/output...
  * **MINOR version (middle number)**: add functionality or changes in a
    backwards-compatible manner. Examples: add output field, add non-mandatory
    input parameter, use a different tool that produces same results...
  * **PATCH version (last number)**: changes/bug fixes that do not affect
    the api/interface. Examples: typo fix, change/add warning messages...
* [x] All inputs are used in process.
* [x] All output fields have their ``re-save`` calls.

### Processes ###

* Code that was once accepted should be kept like it is. It should be changed
  only when its functionality has changed. Small parts may be changed
  to fix bugs (by changing code only to make it nicer you might introduce a bug).
* Set the optimal number of cores that the tool can use (check if parallelization
  can be performed).
* Description of tool should be self explanatory with links to tool's homepage
  and publication.
* Write explanatory input labels and add short command line parameter names
  in square brackets (for example: Maximum length [``--maxlength``]).
* Where possible provide default values for the process input fields.
  If they are provided, then ``required: false`` is not needed.
* Add descriptive error and warning messages.
* Place a ``re-checkrc`` call after each process step (call of a tool, a
  Bash command, a Python script or an R script).
* Add ``re-progress`` calls where appropriate.

### Commit and PR messages ###

* Each commit should be minimal (i.e. 1 change) and self-contained (including
  tests).
* Changes to Docker images need to be accepted in a separate pull request.
  Before the new images can be used for automated testing of your PR, they
  must be manually tagged and pushed to Docker Hub.
* Mark incomplete PRs with the [WIP] (Work In Progress) tag.
* Do not paste links to private repositories from the public one.

### Tests ###

* Add meaningful names of test files with spaces:
  * Genomes: ``genome species.fa.gz``
  * Annotations: ``annotation species.gtf.gz``
  * Adapters: ``adapters-name/type-source``.
* Output and specific input files should include the name of the tested tool.
* Keep test files' size small. If there is no other option, add the large
  test file to the ``files/large`` directory so it will be tracked with Git LFS.
* Pay attention to edge cases: what will happen if any data is
  missing, if length of some list is 0, if string is given where one expects
  integer, if file is empty, ...
* When testing workflows, try to verify workflow success by checking one
  of the last steps. Pick the one that is most likely to fail. Also test
  that all steps have 'OK' status.

### CODE REVIEW

File: resolwe_bio/processes/support_processors/insert_size.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    Tool from Picard, wrapped by GATK4. See GATK
+    CollectInsertSizeMetrics for more information.
+    """
+
+    slug = 'insert-size'
+    name = 'InsertSizeMetrics'
+    process_type = 'data:picard:insert'
+    version = '1.0.0'
+    shaduling_class = SchedulingClass.BATCH
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/insert_size.py]: Typo in scheduling, see #861.

File: resolwe_bio/processes/support_processors/insert_size.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        genome = DataField('genome:fasta', label='Genome')
+        minimum_fraction = FloatField(
+            label='Minimum fraction of reads in a category to be considered',
+            description='When generating the Histogram, discard any data'
+            'categories (out of FR, TANDEM, RF) that have fewer than this'
+            'fraction of overall reads. (Range: 0 to 1).',
+            default=0.05
+        )
+        include_duplicates = BooleanField(
+            label='Include reads marked as duplicates in the insert size histogram.',
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/insert_size.py]: No period for labels.

File: resolwe_bio/processes/support_processors/insert_size.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    data_name = '{{ bam|sample_name|default("?") }}'
+
+    class Input:
+        """Input fields for InsertSizeMetrics."""
+
+        bam = DataField('alignment:bam', label='Alignment BAM file')
+        genome = DataField('genome:fasta', label='Genome')
+        minimum_fraction = FloatField(
+            label='Minimum fraction of reads in a category to be considered',
+            description='When generating the Histogram, discard any data'
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/insert_size.py]: Is Histogram of any special importance that it has a capital H?

File: resolwe_bio/processes/support_processors/insert_size.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            'categories (out of FR, TANDEM, RF) that have fewer than this'
+            'fraction of overall reads. (Range: 0 to 1).',
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/insert_size.py]: Period after reads not necessary.
Should Range be capitalized?

File: resolwe_bio/processes/support_processors/insert_size.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            default=0.05
+        )
+        include_duplicates = BooleanField(
+            label='Include reads marked as duplicates in the insert size histogram.',
+            default=False
+        )
+        deviations = FloatField(
+            label='Deviations limit',
+            description='Generate mean, sd and plots by trimming the data down'
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/insert_size.py]: `sd` could be expanded to "standard deviation". See also line 60.

File: resolwe_bio/processes/support_processors/insert_size.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    }
+    data_name = '{{ bam|sample_name|default("?") }}'
+
+    class Input:
+        """Input fields for InsertSizeMetrics."""
+
+        bam = DataField('alignment:bam', label='Alignment BAM file')
+        genome = DataField('genome:fasta', label='Genome')
+        minimum_fraction = FloatField(
+            label='Minimum fraction of reads in a category to be considered',
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/insert_size.py]: Multiline strings may need an extra space at the end (i.e. see line 66) to properly format spaces between words.

File: resolwe_bio/processes/support_processors/insert_size.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+            choices=[
+                ('STRICT', 'STRICT'),
+                ('LENIENT', 'LENIENT'),
+                ('SILENT', 'SILENT')
+            ],
+            default='STRICT',
+        )
+        assume_sorted = BooleanField(
+            label='Sorted BAM file',
+            description='If true the sort order in the header file will be ignored. Default is False.',
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/insert_size.py]: true -> `True`

Add comma after `True`.

**Comment by AnzeLovse** [comment]
Some things still need to be resolved. One is that MultiQC supports CollectWgsMetrics from version 1.7 onward. Do we plan to upgrade that in the near future? Also there is a slight issue with parsing report file [https://github.com/ewels/MultiQC/issues/966](see). 
Second, I have to add meaningful test files for rrbs-metrics process.

**Comment by jkokosar** [comment]
@AnzeLovse We are currently using a fork of MultiQC 1.8.0 (for some other, now already merged fixes), and plan to upgrade to the official 1.9 when released.

File: resolwe_bio/processes/support_processors/insert_size.py [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        basename = os.path.basename(inputs.bam.bam.path)
+        assert basename.endswith('.bam')
+        name = basename[:-4]
+        metrics_file = f'{name}_insert_size_metrics.txt'
+        histogram_file = f'{name}_insert_size.pdf'
+
+        args = [
+            '--INPUT', inputs.bam.bam.path,
+            '--OUTPUT', metrics_file,
+            '--Histogram_FILE', histogram_file,
```
[the remark this answers is not in the document]
Comment (AnzeLovse) [reply resolwe_bio/processes/support_processors/insert_size.py]: Unfortunately this is case sensitive.

File: resolwe_bio/processes/support_processors/alignment_summary.py [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+            '--ASSUME_SORTED', inputs.assume_sorted,
+            '--EXPECTED_PAIR_ORIENTATIONS', inputs.pair_orientation,
+            '--IS_BISULFITE_SEQUENCED', inputs.bisulfite,
+        ]
+
+        if inputs.adapters:
+            adapters_list = get_sequences(inputs.adapters.fasta.path)
+            args.extend(['--ADAPTER_SEQUENCE', adapters_list])
+        else:
+            # Clear the default adapter list implemeted in Picard.
```
Comment (jkokosar) [inline resolwe_bio/processes/support_processors/alignment_summary.py]: implemeted -> implemented

File: resolwe_bio/processes/support_processors/insert_size.py (hunk 2 above)
Comment (jkokosar) [reply resolwe_bio/processes/support_processors/insert_size.py]: @AnzeLovse please remove the period

File: docs/CHANGELOG.rst [hunk 10]
```diff
@@ -49,7 +53,6 @@ Fixed
 - Fix Mapping search for ``source_id`` / ``target_id``
 
 
-===================
```
Comment (jkokosar) [inline docs/CHANGELOG.rst]: @AnzeLovse don't remove this line.
